data/lib/release/changelog_gen.py

```
@@ -0,0 +1,197 @@
+
+
+def _get_commit_list() -> list[str]:
+    from data.lib.release.git_util import find_last_release_tag
+
+    last_tag = find_last_release_tag()
+    if last_tag:
+        cmd = ["git", "log", "--oneline", "--no-decorate", f"{last_tag}..HEAD"]
+    else:
+        cmd = ["git", "log", "--oneline", "--no-decorate"]
+
+    result = subprocess.run(cmd, capture_output=True, text=True, cwd=PROJECT_ROOT)
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)